app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -223,9 +224,21 @@ public static void playOnExternalVideoPlayer(final Context context, final Stream
 
     public static void playOnExternalPlayer(final Context context, final String name,
                                             final String artist, final Stream stream) {
+        final String url;
+        final DeliveryFormat deliveryFormat = stream.getDeliveryFormat();
+        if (deliveryFormat instanceof DeliveryFormat.Direct) {
+            url = ((DeliveryFormat.Direct) deliveryFormat).getUrl();
+        } else if (deliveryFormat instanceof DeliveryFormat.HLS) {
+            url = ((DeliveryFormat.HLS) deliveryFormat).getUrl();
+        } else {
+            Toast.makeText(context, R.string.selected_stream_external_player_not_supported,
```

Comment:
Yes, that's right.

Although, we could send instead the url pointing to the DASH manifest containing all the resolutions, but it'd make the resolution picker a little useless and maybe confusing.

Maybe even saving a file containing a manually crafted video+audio manifest and sending that to external players? I think it could work.

Reply:
I assume such videos would throw an error if shared from the Youtube app to VLC, for example?